src/dotnet-setversion/Program.cs

```diff
@@ -96,6 +113,27 @@ private static void SetVersion(string version, string csprojFile)
                                   .SingleOrDefault() ?? projectNode
                                   .GetOrCreateElement("PropertyGroup")
                                   .GetOrCreateElement("Version");
+            if (version.StartsWith("@"))
+            {
+                // Read version number from a file
+                var filename = version.Substring(1);
+                if (!File.Exists(filename))
+                {
+                    throw new FileNotFoundException("The filename specified in the version parameter was not found", filename);
+                }
+
+                var versionFileText = File.ReadAllText(filename);
+                if (versionFileText.StartsWith("{"))
```

Comment:
Instead of checking if it starts with a brace, I'd try to deserialise it as JSON and fall back to interpreting it as a simple string if a JSON Exception is thrown.

This should make it more robust to lexically insignificant features of the file e.g. starting with a blank line.